core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -75,8 +75,11 @@ trait Runtime[+R] {
    * Runs the effect "purely" through an async boundary. Useful for testing.
    */
   final def run[E, A](zio: ZIO[R, E, A])(implicit trace: ZTraceElement): IO[E, A] =
-    IO.async[E, A] { callback =>
-      unsafeRunAsyncWith(zio)(exit => callback(ZIO.done(exit)))
+    ZIO.fiberId.flatMap { fiberId =>
+      IO.asyncInterrupt[E, A] { callback =>
+        val canceler = unsafeRunAsyncCancelable(zio)(exit => callback(ZIO.done(exit)))
+        Left(ZIO.succeed(canceler(fiberId)))
```

Comment:
`canceler` is blocking so I believe this needs to be `ZIO.blocking` instead of `ZIO.succeed`. Sorry about that!